go/worker/storage/committee/node.go

```diff
@@ -871,15 +882,32 @@ func (n *Node) worker() { // nolint: gocyclo
 	}
 
 	// Try to perform initial sync from state and io checkpoints.
-	if !n.checkpointSyncDisabled {
+	if !n.checkpointSyncDisabled || n.checkpointSyncForced {
 		var summary *blockSummary
-		summary, err = n.syncCheckpoints()
-		if err != nil {
-			// Try syncing again. The main reason for this is the sync failing due to a
-			// checkpoint pruning race condition (where nodes list a checkpoint which is
-			// then deleted just before we request its chunks). One retry is enough.
-			n.logger.Info("first checkpoint sync failed, trying once more", "err", err)
+	CheckpointSyncRetry:
+		for {
 			summary, err = n.syncCheckpoints()
+			switch n.checkpointSyncForced {
+			case true:
+				// We have no other options but to perform a checkpoint sync as we are missing
+				// genesis state.
+				n.logger.Info("checkpoint sync required as we don't have genesis state, retrying")
```

Comment:
Should this also log the error returned from ` n.syncCheckpoints`? (which is currently not logged in case `n.checkpointSyncForced=true`).

Also won't this currently just loop forever in case `n.checkpointSyncForced` is true? I don't see anything that ever sets `n.checkpointSyncForced` to false? Should this be within `if err != nil {`?

Reply:
Yeah it should be, I'm just adding a test for this.